### ISSUE TITLE
Broken eq/neq

### ISSUE THREAD

**Issue by JajaComp** [JetBrains/Exposed#1489]
After update to version 0.38.1 i got issue:
I have field in entity:
`val object_id = reference("object_id", ObjectEntity).nullable()`

and now i can't correct check this:
```
val objectId: UUID? = null
val where1 = Op.build { _LocalRoleEntity.object_id eq objectId }
val where2 = Op.build { _LocalRoleEntity.object_id neq objectId }
```

**Comment by Tapac** [comment]
Is the problem the same if you'll replace your reference with `optReference("object_id", ObjectEntity)` ?

**Comment by JajaComp** [comment]
This does not change anything. 
```
Type mismatch.
Required:
TypeVariable(T)
Found:
UUID?
```

In version 0.37.3 all fine

**Comment by Tapac** [comment]
will be fixedin the next release

**Comment by Tapac** [comment]
@ESchouten, oh, my bad. It was not so easy to find the way to fix for eq and I forgot about neq.
Thank you for notice.

**Comment by mark-sinclair-tw** [comment]
I was attempting to upgrade from 0.37.3 to 0.38.2, and noticed I was getting new compiler errors when comparing a database ID to a nullable `Int?` with `neq`:

```kotlin
val itemID: Int? = 5
Items.select { Items.id neq itemID }
```
fails to compile with `Type mismatch: inferred type is Int? but TypeVariable(T) was expected`

This was marked as fixed in the April 19 `0.38.2` release here: https://github.com/JetBrains/Exposed/blob/master/docs/ChangeLog.md#0382

But the commit that actually fixed it did not make it into that release, and only happened on April 20: https://github.com/JetBrains/Exposed/commit/b456bbe9499cb72feaa0c1ac8af8a3cbcfd73603#diff-3c3e7274f9f251b7e17658c0519a0852fe3b698401e13203c8f19828a8612e68 

So we have to hold back upgrading until the next release is out

**Comment by Tapac** [comment]
@mark-sinclair-tw , sorry for a delay with the release. It will be on the next week.
